rcl/src/rcl/node.c

```diff
@@ -99,7 +99,20 @@ const char * rcl_create_node_logger_name(
   return node_logger_name;
 }
 
-const char * rcl_get_secure_root(const char * node_name, const rcl_allocator_t * allocator)
+/// Return the secure root dir associated with a node given the validated node name and namespace.
+/**
+ * E.g. for a node named "c" in namespace "/a/b", the secure root path will be
+ * "a/b/c", where the delimiter "/" is normlized for target file system (e.g. "\\" for _WIN32).
+ *
+ * \param[in] node_name validated node name (a single token)
+ * \param[in] node_namespace validated, absolute namespace (starting with "/")
+ * \param[in] allocator the allocator to use for allocation
+ * \returns duplicated string or null if there is an error
```

Comment:
what does `duplicated string` means in this context?
suggestions:
"machine specific node secure root path" would be more descriptive?
or simply "(absolute) node secure root path" maybe

Reply:
Forgotten copypasta from `rcl_create_node_logger_name`; combined your phrasing for docblock.